v03_pipeline/lib/reference_data/dataset_table_operations.py

```diff
@@ -173,6 +177,113 @@ def annotate_dataset_globals(joined_ht: hl.Table, dataset: str, dataset_ht: hl.T
     )
 
 
+def validate_joined_ht_globals_match_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    reference_genome: ReferenceGenome,
+) -> bool:
+    config = CONFIG[dataset][reference_genome.v02_value]
+    checks = {
+        'version': _ht_version_matches_config(
+            joined_ht,
+            dataset,
+            config,
+            reference_genome,
+        ),
+        'path': _ht_path_matches_config(joined_ht, dataset, config),
+        'enum': _ht_enums_match_config(joined_ht, dataset, config),
+        'select': _ht_selects_match_config(joined_ht, dataset, config),
+    }
+
+    results = []
+    for check, result in checks.items():
+        if result is False:
+            logger.info(f'{check} mismatch for {dataset}')
+
+        results.append(result)
+
+    return all(results)
+
+
+def _ht_version_matches_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    dataset_config: dict,
+    reference_genome: ReferenceGenome,
+) -> bool:
+    joined_ht_version = hl.eval(
```

Comment:
One quick hail performance tip.. you want t run `hl.eval` as new times as possible.  So my guess is that writing these functions to take in a `joined_ht`'s evaluated globals is going to be much nicer and faster!

Reply:
I decided to return a `hl.BooleanExpression` from the each of the check functions and call `hl.eval()` on the result of each of those functions, but I'll see if I can get away with calling it once on joined_ht's evaluated globals